corgi/core/models.py

```diff
@@ -923,19 +861,51 @@ def get_product_details(variant_names: tuple[str], stream_names: list[str]) -> d
 class ComponentQuerySet(models.QuerySet):
     """Helper methods to filter QuerySets of Components"""
 
-    def released_components(self, include: bool = True) -> models.QuerySet["Component"]:
+    def filter_latest_nevra_by_name(self, component_name: str) -> str:
+        nevras = self.filter(name=component_name).values_list("nevra", flat=True)
+        if nevras:
+            # Get the latest NVR using python. This only works for valid RPM NEVRAs
+            return sorted(nevras, key=functools.cmp_to_key(compare_packages))[-1]
```

Comment:
I guess this may not work so well for components with versions that are just commit hashes but let's file that as a separate issue.